src/design-system/components/Box/Box.tsx

```diff
@@ -286,31 +285,47 @@ export const Box = forwardRef(function Box(
   );
 }) as PolymorphicBox;
 
-function useShadow({
-  shadow: shadowProp,
-  shadowColor: shadowColorProp = 'shadow',
-}: {
-  shadow: BoxProps['shadow'];
-  shadowColor: BoxProps['shadowColor'];
-}) {
-  const shadow = resolveToken(shadows, shadowProp);
+function useShadow(shadowProp: BoxProps['shadow']) {
+  const { shadowVariant, shadowColor } = useMemo(() => {
+    let shadowVariant;
+    let shadowColor: ShadowColor | CustomColor = 'shadow';
 
-  let shadowColors = useMemo(
-    () =>
-      Array.isArray(shadowColorProp)
-        ? [...shadowColorProp].reverse()
-        : [shadowColorProp],
-    [shadowColorProp]
-  );
-  const parsedShadowColors = useForegroundColors(shadowColors);
+    if (typeof shadowProp === 'string') {
+      const shadowMatcher = shadowProp.match(/^(\d*px\s\w+)\s?(\w+$)?/);
```

Comment:
I feel like we should generate all the standard shadow values up front so at runtime it's a simple object lookup rather than having to parse the string. This also makes it easier to restrict which values are allowed without relying on TypeScript.